tests/func/utils/test_hydra.py

```diff
@@ -167,16 +169,69 @@ def hydra_setup(tmp_dir, config_dir, config_name):
         ),
     ],
 )
-def test_compose_and_dump(tmp_dir, suffix, overrides, expected):
+def test_compose_and_dump_overrides(tmp_dir, suffix, overrides, expected):
     from dvc.utils.hydra import compose_and_dump
 
     config_name = "config"
     output_file = tmp_dir / f"params.{suffix}"
     config_dir = hydra_setup(tmp_dir, "conf", "config")
-    compose_and_dump(output_file, config_dir, config_name, overrides)
+    config_module = None
+    compose_and_dump(output_file, config_dir, config_module, config_name, overrides)
     assert output_file.parse() == expected
 
 
+def hydra_setup_dir_basic(tmp_dir, config_subdir, config_name, config_content):
+    if config_subdir is None:
+        return None
+
+    config_dir = tmp_dir / config_subdir
+    config_dir.mkdir()
+    (config_dir / f"{config_name}.yaml").dump(config_content)
+    return str(config_dir)
+
+
+@pytest.mark.parametrize(
+    "config_subdir,config_module,config_name,error_context,config_content",
+    [
+        ("conf", None, "config", does_not_raise(), {"normal_yaml_config": True}),
+        (
+            None,
+            "hydra.test_utils.configs",
+            "config",
+            does_not_raise(),
+            {"normal_yaml_config": True},
+        ),
+        (
+            "conf",
+            "hydra.test_utils.configs",
+            "config",
+            pytest.raises(ValueError, match="^Exactly one of"),
+            None,
+        ),
+        (
+            None,
+            None,
+            "config",
+            pytest.raises(ValueError, match="^Exactly one of"),
+            None,
+        ),
+    ],
+)
+def test_compose_and_dump_dir_module(
+    tmp_dir, config_subdir, config_module, config_name, error_context, config_content
+):
+    from dvc.utils.hydra import compose_and_dump
+
+    output_file = tmp_dir / "params.yaml"
+    config_dir = hydra_setup_dir_basic(
```

Comment:
It would be nice if we could dump a different content to `config_dir` and only do it if `config_dir` is not None, to explicitly verify that we are using `initialize_config_dir` or `initialize_config_module`.